### PR TITLE
Add endpoints for error report

### PR DESCRIPTION
**Pull request by SemyonKorostik**

### CODE REVIEW

File: src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
-    public ErrorReportsController(ErrorReportsService errorReportsService) {
+    public ErrorReportsController(ErrorReportsService errorReportsService,
+                                  ErrorReportsConverter errorReportsConverter) {
         this.errorReportsService = errorReportsService;
+        this.errorReportsConverter = errorReportsConverter;
+    }
+
+    @ApiOperation(value = "get a list of error reports by test id")
+    @GetMapping(path = "/{testId}")
+    public List<ErrorReportDTO> getByTestId(@PathVariable long testId) {
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java]: Why do we really need a separate endpoint for it?

File: src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java [hunk 2]
```diff
@@ -1,25 +1,42 @@
...
     @Autowired
-    public ErrorReportsController(ErrorReportsService errorReportsService) {
+    public ErrorReportsController(ErrorReportsService errorReportsService,
+                                  ErrorReportsConverter errorReportsConverter) {
         this.errorReportsService = errorReportsService;
+        this.errorReportsConverter = errorReportsConverter;
+    }
+
+    @ApiOperation(value = "get a list of error reports by test id")
+    @GetMapping(path = "/{testId}")
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java]: Do anyone going to use the endpoint?

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

File: src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java (hunk 1 above)
Comment (SemyonKorostik) [reply src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java]: It was necessary to get a list of error reports. I thought that it meant getting them by the test id. By what parameter should I have got a list of error reports?

Comment (AntonMaksimenkau) [reply src/main/java/com/team4/testingsystem/controllers/ErrorReportsController.java]: Put the data to a corresponding dto. We don't need to have a separate endpoint for it

File: src/main/java/com/team4/testingsystem/services/ErrorReportsService.java [hunk 3]
```diff
@@ -11,4 +12,6 @@
     void add(String requestBody, Long questionId, Long testId);
 
     void removeByTestAndQuestion(long testId, long questionId);
+
+    void addAll(List<ErrorReport> errorReports);
```
Comment (AntonMaksimenkau) [inline src/main/java/com/team4/testingsystem/services/ErrorReportsService.java]: is not relevant anymore

**Review by AntonMaksimenkau — CHANGES_REQUESTED** [review]

**Review by AntonMaksimenkau — APPROVED** [review]

**Review by maxim12321 — APPROVED** [review]

**Pull request merged**